lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -116,33 +175,34 @@ def import_kymotrackgroup_from_csv(filename, kymo, channel, delimiter=";"):
     """
 
     # TODO: File format validation could use some improvement
-    try:
-        data = np.loadtxt(filename, delimiter=delimiter)
-    except ValueError:  # Could not convert to float
-        raise IOError("Invalid file format!")
+    data, pylake_version, csv_version = _read_txt(filename, delimiter)
 
-    if data.ndim != 2 or data.shape[0] <= 2:
+    mandatory_fields = ["time (pixels)", "coordinate (pixels)"]
+    if not all(f in data for f in mandatory_fields):
         raise IOError("Invalid file format!")
 
-    indices = data[:, 0]
-    time_idx = data[:, 1]
-    tracks = np.unique(indices)
+    for t in data["time (pixels)"]:
```

Comment:
Since `data["time (pixels)"]` contains a list of coordinates per track. We're iterating over the tracks here. I can make that a bit more explicit by calling it `track_time`.